aiutils/tftools/train.py

```diff
@@ -0,0 +1,104 @@
+""" This module defines a hepler class for multiple rate optimizers in Tensorflow
+
+The MultiRateOptimizer class provides a slightly simpler version of the tf.train.Optimizer API
+ - compute_gradients(loss)
+ - apply_gradients(grads_and_vars, global_step)
+ - minimize(loss, global_step)
+
+To add variables and corresponding optimizers use
+ - add_variables(variables, optimizer)
+
+Usage:
+    a = tf.Variable(1)
+    b = tf.Variable(2)
+    c = tf.Variable(3)
+
+    loss = f(a,b,c)
+
+    optimizer = MultiRateOptimizer()
+    optimizer.add_variables([a,b], tf.train.GradientDescentOptimizer(.1))
+    optimizer.add_variables([c], tf.train.GradientDescentOptimizer(.01))
+
+    min_opt = optimizer.minimize(loss)
+
+    min_opt.eval()
+"""
+
+import tensorflow as tf
+import itertools
+
+
+class MultiRateOptimizer():
+    """ Class for managing a multi-rate optimization problem """
+
+    def __init__(self):
+        self.optimizers = []
+        self.variables = []
+
+    def add_variables(self, variables, optimizer):
```

Comment:
The reason I didn't do that is because optimizers have various required parameters, not just learning rate, so it would have to be optimizer_params .  Also, I could potentially see applying different optimizers to different parts of a network.  Eg.  Adam to top (new) layers and gradient descent to lower (fine-tune) layers.

I could change this such that the init takes an optional optimizer constructor, and then add_variables takes either optimizer or param_list.

I think add_variables should probably also return self so that you can write code as

`MultiRateOptimizer(tf.train.GradientDescentOptimizer).add_variables([..], params={'step_len',.1}).add_variables([...], params={'step_len', .01})`